New Search API & Service

jdrossl opened the pull request:
https://github.com/craftercms/craftercms/issues/2787

jdrossl committed:
New Search API

jdrossl committed:
Updates for new Search API

jdrossl committed:
Updates for new Search API:
- Search endpoint now uses a JSON body
- Implemented query & filters for search
- Code cleanup & javadocs

jdrossl committed:
Updates for new Search API:
- Add OAS docs
- Add javadocs

sumerjabri commented on src/main/api/2/studio-api-2.yaml:
```diff
@@ -1321,6 +1356,85 @@ components:
+          description: The order to sort the results (defaults to DESC)
+        filters:
+          type: object
+          description: Map containing field names and values to filter the results
```
Please explain how these are populated. Per our discussion, the caller may use the returned facets.

sumerjabri commented on src/main/api/2/studio-api-2.yaml:
```diff
@@ -1321,6 +1356,85 @@ components:
+          description: List of facets based on the results found
+          items:
+            $ref: '#/components/schemas/SearchFacet'
+    SearchItem:
```
Please rename to SearchResultItem

sumerjabri commented on src/main/resources/crafter/studio/studio-config.yaml:
```diff
@@ -511,3 +511,91 @@ studio.cors.methods: '*'
+    matches: ^/component/.*$
+  - name: Taxonomy
+    field: content-type
+    matches: ^/taxonomy$
```
Please add MS Office docs.

jdrossl committed:
Updates according to feedback:
- Refactor for search related classes
- Added missing type mappings
- Removed GET method support from search controller

jdrossl committed:
Fix type mappings
